File: Domain/Employee.cs
using System;
namespace Domain
{
    public class Employee
    {

        public Guid Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string FullAddress { get; set; }
        public string MaillingAdress { get; set; }
        public string Email { get; set; }
        public string PhoneNumber { get; set; }
        public string CitizenshipStatus { get; set; }
        public DateTime EmploymentStartDate { get; set; }
        public string EmploymentType { get; set; }
        public string Position { get; set; }

        public string EmergencyContactName { get; set; }
        public string EmergencyContactRelationship { get; set; }
        public string EmergencyConteactPhoneNUmber { get; set; }

        public bool IsSigned { get; set; }


    }
}

File: clientapp/src/app/models/employee.ts

export interface Employee{
       id: string;
        firstName: string;
        lastName: string;
        fullAddress: string;
        maillingAdress: string;
        email: string;
        phoneNumber: string;
        citizenshipStatus: string;
        employmentStartDate: Date;
        employmentType: string;
        position: string;
        emergencyContactName: string;
        emergencyContactRelationship: string;
        emergencyConteactPhoneNUmber: string;
        isSigned: boolean;
}

export class Employee implements Employee{
        id = '';
        firstName = "";
        lastName = "";
        fullAddress = "";
        maillingAdress = "";
        email = "";
        phoneNumber = "";
        citizenshipStatus = "";
        employmentStartDate = new Date();
        employmentType = "";
        position = "";
        emergencyContactName = "";
        emergencyContactRelationship = "";
        emergencyConteactPhoneNUmber = "";
        isSigned = false;

        constructor(emp?:Employee){
                if(emp){
                        this.id = emp.id;
                        this.firstName = emp.firstName
                        this.lastName = emp.lastName;
                        this.fullAddress = emp.fullAddress;
                        this.maillingAdress = emp.maillingAdress;
                        this.email = emp.email;
                        this.phoneNumber = emp.phoneNumber;
                        this.citizenshipStatus = emp.citizenshipStatus;
                        this.employmentStartDate = new Date( emp.employmentStartDate);
                        this.employmentType = emp.employmentType;
                        this.position = emp.position;
                        this.emergencyContactName = emp.emergencyContactName;
                        this.emergencyContactRelationship = emp.emergencyContactRelationship;
                        this.emergencyConteactPhoneNUmber = emp.emergencyConteactPhoneNUmber;
                        this.isSigned = emp.isSigned;
                }
                

        }
        
}

